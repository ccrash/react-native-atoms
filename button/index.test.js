import React from 'react'
import renderer from 'react-test-renderer'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Button from './index'

configure({ adapter: new Adapter() })

describe('Testing Button component', () => {
  const submitCallback = jest.fn()

  const wrapper = shallow(<Button onPress={submitCallback} />)

  it('Renders a button with text passed as props', () => {
    expect(renderer.create(<Button title={'Hello world'} />)).toMatchSnapshot()
  })

  it('Renders a button with text passed as children', () => {
    expect(renderer.create(<Button>{'Hello world'}</Button>)).toMatchSnapshot()
  })

  it('Renders a button with custom component passed as children', () => {
    expect(
      renderer.create(
        <Button>
          <Button>{'Hello world'}</Button>
        </Button>,
      ),
    ).toMatchSnapshot()
  })

  it('calls actions as expected when clicked', () => {
    wrapper.setProps({ value: 'test' })
    wrapper
      .find('TouchableOpacity')
      .first()
      .simulate('press')
    expect(submitCallback).toBeCalledWith('test')
  })
})
