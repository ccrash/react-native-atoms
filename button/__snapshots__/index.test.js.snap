// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component Renders a button with custom component passed as children 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#CCCCCC",
      "borderRadius": 10,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 5,
      "opacity": 1,
      "padding": 20,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#CCCCCC",
        "borderRadius": 10,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 5,
        "opacity": 1,
        "padding": 20,
        "width": "100%",
      }
    }
  >
    Hello world
  </View>
</View>
`;

exports[`Testing Button component Renders a button with text passed as children 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#CCCCCC",
      "borderRadius": 10,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 5,
      "opacity": 1,
      "padding": 20,
      "width": "100%",
    }
  }
>
  Hello world
</View>
`;

exports[`Testing Button component Renders a button with text passed as props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#CCCCCC",
      "borderRadius": 10,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 5,
      "opacity": 1,
      "padding": 20,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#404040",
          "fontSize": 22,
          "fontWeight": "600",
        }
      }
    >
      Hello world
    </Text>
  </View>
</View>
`;
